2018AsianGamesApp/app/src/main/java/com/example/mrtbkwjn/a2018asiangamesapp/LoginActivity.java

```diff
@@ -1,45 +1,186 @@
-        // Hide the status bar.
-        //int uiOptions = View.SYSTEM_UI_FLAG_FULLSCREEN;
-       // decorView.setSystemUiVisibility(uiOptions);
-        // Remember that you should never show the action bar if the
-        // status bar is hidden, so hide that too if necessary.
-       // ActionBar actionBar = getActionBar();
-        //actionBar.hide();
+    private void firebaseAuthWithGoogle(GoogleSignInAccount acct)   {
+        Log.d(TAG, "firebaseAuthWithGoogle:" + acct.getId());
+        // [START_EXCLUDE silent]
+        showProgressDialog();
+        // [END_EXCLUDE]
 
+        AuthCredential credential = GoogleAuthProvider.getCredential(acct.getIdToken(), null);
+        mobileAuth.signInWithCredential(credential)
+                .addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
+                    @Override
+                    public void onComplete(@NonNull Task<AuthResult> task) {
+                        if (task.isSuccessful()) {
+                            // Sign in success, update UI with the signed-in user's information
+                            Log.d(TAG, "Sign-in Success");
+                            FirebaseUser user = mobileAuth.getCurrentUser();
+                            updateUI(user);
+                        } else {
+                            // If sign in fails, display a message to the user.
+                            Log.w(TAG, "Sign-in Failed", task.getException());
+                            Snackbar.make(findViewById(R.id.activity_login), "Authentication Failed.", Snackbar.LENGTH_SHORT).show();
+                            updateUI(null);
+                        }
+
+                        // [START_EXCLUDE]
+                        hideProgressDialog();
+                        // [END_EXCLUDE]
+                    }
+                });
+    }
+
+    // [START signin]
+    private void signIn() {
+        Intent signInIntent = mGoogleSignInClient.getSignInIntent();
+        startActivityForResult(signInIntent, RC_SIGN_IN);
+    }
+    // [END signin]
+
+
+    private void onConnectionFailed (ConnectionResult connectionResult){
+        Log.d(TAG, "onConnectionFailed" + connectionResult);
+    }
+
+
+    private void revokeAccess() {
+        // Firebase sign out
+        mobileAuth.signOut();
+
+        // Google revoke access
+        mGoogleSignInClient.revokeAccess().addOnCompleteListener(this,
+                new OnCompleteListener<Void>() {
+                    @Override
+                    public void onComplete(@NonNull Task<Void> task) {
+                        updateUI(null);
+                    }
+                });
+    }
+
+    private void updateUI(FirebaseUser user) {
+        hideProgressDialog();
+        if (user != null) {
+
+            findViewById(R.id.googleLoginButton).setVisibility(View.GONE);
+
+        } else {
+
+            findViewById(R.id.googleLoginButton).setVisibility(View.VISIBLE);
+        }
     }
 
```

Comment:
too many empty line here, consider using only 1 or 2 empty line to make the code more compact